.github/workflows/bump_example_version.yml

```diff
@@ -0,0 +1,37 @@
+name: Bump Action version
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  update-version:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+        with:
+          fetch-depth: 0
+
+      - name: Update dirty-waters-action version
+        run: |
+          COMMIT_SHA=$(git rev-parse HEAD)
+
+          # Use sed to update occurrences of the version in the workflow file
+          sed -i "s/chains-project\/dirty-waters-action@v[0-9.]\+/chains-project\/dirty-waters-action@${COMMIT_SHA}/g" ./example_workflow.yml
+
+          echo "commit_sha=$COMMIT_SHA" >> $GITHUB_ENV
+
+      - name: Update dirty_waters_version in the action to the repo's latest tag
+        run: |
+          LATEST_TAG=$(git ls-remote --tags https://github.com/chains-project/dirty-waters.git | awk -F/ '{print $3}' | sort -V | tail -n1)
+          sed -i "s/DIRTY_WATERS_VERSION=\".*\"/DIRTY_WATERS_VERSION=\"$LATEST_TAG\"/" action.yml
+
+      - name: Commit changes
+        uses: stefanzweifel/git-auto-commit-action@v5.1.0
```

Comment:
Given that `actions/checkout` is pinned to a commit SHA, I think it would make sense to do the same here